pkg/controller/machine.go

```diff
@@ -566,11 +566,22 @@ func (c *controller) machineDelete(machine *v1alpha1.Machine, driver driver.Driv
 			return err
 		}
 
+		// Remove finalizers from machine object
 		c.deleteMachineFinalizers(machine)
```

Comment:
Shouldn't we do this until after node is deleted? To make sure reconciliation can take care of failed node deletion?

Reply:
Yes. My bad. I shall do it the other way.

Reply:
Fixed it here - https://github.com/gardener/machine-controller-manager/pull/316/commits/b7b6b395be405ccf6a899bf19cbc88370103884d